sqlx-mysql/src/arguments.rs

```diff
@@ -22,9 +23,11 @@ impl MySqlArguments {
         self.types.push(ty);
         self.null_bitmap.resize((index / 8) + 1, 0);
 
-        if let IsNull::Yes = value.encode(&mut self.values) {
+        if let IsNull::Yes = value.encode(&mut self.values)? {
```

Comment:
To avoid leaving the buffer in an invalid state, this should use a RAII guard that rolls back changes to it if this method returns early.

Reply:
Fixed for MySql.